core/client/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -66,15 +66,30 @@
   }
 
   /**
-   * Acquires a block master client from the block master client pool.
+   * Acquires a file system master client from the file system master client pool.
    *
-   * @return the acquired block master client
+   * @return the acquired file system master client
    */
   public FileSystemMasterClient acquireMasterClient() {
     return mFileSystemMasterClientPool.acquire();
   }
 
   /**
+   * Acquires a file system master client from the file system master client pool. The resource is
+   * {@code Closeable}.
+   *
+   * @return the acquired file system master client resource
+   */
+  public CloseableResource<FileSystemMasterClient> acquireMasterClientResource() {
```

Comment:
I agree. The current solution is not ideal. I am thinking to add a new method e.g. `AsSharedResource()`  in `Client` base class, which implements `close` by returning itself to the pool. That way it might be clear and no need to use anonymous class.